FinancialTracker/Firebase.swift

```diff
@@ -161,6 +172,34 @@ class FirebaseHandler {
         self.user.balance = balance
         
         firestore.collection(usersKey).document(currentUser.uid).setData([balanceKey: balance], merge: true)
+        
+        completionHandler(nil, true)
+    }
+    
+    func addExpenseToCurrentUser(_ expenseAmount: Int, category: String, completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
+        guard let currentUser = currentUser else {
+            completionHandler(FirebaseError.access, false)
```

Comment:
Looks like the `access` error is quite often connected with the `currentUser`, maybe, you should think about a better naming for it. Else, you can add have the `access` case have an associated string and add more description for each case where the `access` error case is used.